src/bit-systems/scene-loading.ts

```diff
@@ -81,7 +90,23 @@ function* loadScene(
     }
     environmentSystem.updateEnvironmentSettings(envSettings);
 
-    const sceneEl = AFRAME.scenes[0];
+    const cameraNode = (document.getElementById("scene-preview-node") as AElement).object3D!;
+    removeComponent(world, ScenePreviewCamera, cameraNode.eid!);
+    const sceneObj = world.eid2obj.get(scene)!;
+    const previewCamera = sceneObj.getObjectByName("scene-preview-camera");
+    if (previewCamera) {
+      cameraNode.position.copy(previewCamera.position);
+      cameraNode.rotation.copy(previewCamera.rotation);
+      cameraNode.rotation.reorder("YXZ");
+    } else {
+      const cameraPos = cameraNode.position;
+      cameraNode.position.set(cameraPos.x, 2.5, cameraPos.z);
+    }
+    cameraNode.matrixNeedsUpdate = true;
+    addComponent(world, ScenePreviewCamera, cameraNode.eid!);
+    ScenePreviewCamera.duration[cameraNode.eid!] = 60;
+    ScenePreviewCamera.positionOnly[cameraNode.eid!] = 1;
```

Comment:
Probably I just don't understand the code yet... What is this "remove and then add ScenePreviewCamera component" for?

Reply:
This component/system it's a bit special. We don't really add the `scene-preview-camera` component in the glTF, we add a node with `scene-preview-camera` name (not component). When the scene is loading we get that node using `getObjectByName` and we add the `ScenePreviewCamera` component to a pre-exiting scene node (`scene-preview-node`). 

Every time a scene is loaded we add the `ScenePreviewCamera` component to the `scene-preview-node` node. We use the remove/add so we can trigger the exit/enter queries and cleanup the previous preview camera component state.